MetaMask/metamask-extension
# ConfirmDecryptMessage: avoid mutation of state in decryptMsg/completedTx action

## Explanation

Fixes #17867

Explanation copied from https://github.com/MetaMask/metamask-extension/pull/17772:
> In https://github.com/MetaMask/metamask-extension/pull/17508 i[@brad-decker] added store creation using reduxjs/toolkit instead of vanilla redux. One thing that comes with this is detection of mutation in state, which will throw an invariant error. This should only be happening in dev mode but its good to fix. This fixes the specific situation by cloning the transaction object before it used.


## Manual Testing Steps



1. Go to https://metamask.github.io/test-dapp/
2. Submit "Get Encryption Key"
3. Type a message and click "Encrypt"
4. Click "Decrypt". The popup should open.
5. Open the transaction in full-page, home.html
6. Inspect console
7. Click "Decrypt"
8. Observe the transaction is not completed and the error appears in console
9. To bypass this, double-click on "Decrypt" button


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
